feat(linter): Add options for `typescript/require-array-sort-compare` rule.

connorshea opened the pull request:
This option was already implemented on the tsgolint side, this just adds the config to the oxlint side.

See https://github.com/oxc-project/tsgolint/blob/main/internal/rules/require_array_sort_compare/options.go

Part of #16978.

```md
## Configuration

This rule accepts a configuration object with the following properties:

### ignoreStringArrays

type: `boolean`

default: `true`

Whether to ignore arrays in which all elements are strings.
```

camc314 commented:
### Merge activity

* **Dec 17, 9:12 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 17, 9:12 AM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

connorshea committed:
feat(linter): Add options for `typescript/require-array-sort-compare` rule. (#16980)

This option was already implemented on the tsgolint side, this just adds the config to the oxlint side.

See https://github.com/oxc-project/tsgolint/blob/main/internal/rules/require_array_sort_compare/options.go

Part of #16978.

```md
## Configuration

This rule accepts a configuration object with the following properties:

### ignoreStringArrays

type: `boolean`

default: `true`

Whether to ignore arrays in which all elements are strings.
```
